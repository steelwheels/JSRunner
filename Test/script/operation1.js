/*
 * operation0.js
 */

function allocateOperation(opfile, a, b, c)
{
	let url = URL(opfile) ;
	let op  = Operation([url], console) ;
	op.set("a", a) ;
	op.set("b", b) ;
	op.set("c", c) ;
	return op ;
}

function main()
{
	let op0 = allocateOperation("../Test/Script/Operation/op1.js", 1, 2, 3) ;
	let op1 = allocateOperation("../Test/Script/Operation/op1.js", 3, 4, 7) ;

	let queue0 = OperationQueue() ;
	if(!queue0.execute(op0)){
		console.error("[Error] Could not execute op0\n") ;
		return false ;
	}
	if(!queue0.execute(op1)){
		console.error("[Error] Could not execute op1\n") ;
		return false ;
	}
	queue0.waitOperations() ;

	return true ;
}
