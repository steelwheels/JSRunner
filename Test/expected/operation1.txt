[op0] program
[op1] program
[op0] start
[op1] start
[opX] [begin] wait operations
[opX] main
[opX] main
[opX] [end  ] wait operations
